usr/src/lib/libzfs_core/common/libzfs_core.c

```diff
@@ -203,6 +204,23 @@ lzc_clone(const char *fsname, const char *origin,
 	return (error);
 }
 
+int
+lzc_promote(const char *fsname, const char *origin)
```

Comment:
Thanks for the feedback.

I agree with Matt.  Please hold off integrating this change while I look at how easy it would be to fix the ioctl.